python/metatensor-core/metatensor/tensor.py

```diff
@@ -732,12 +732,19 @@ def _list_or_str_to_array_c_char(strings: Union[str, Sequence[str]]):
     return c_strings
 
 
+def _is_numpy_int(value):
+    return np.can_cast(value, np.int32, casting="same_kind")
+
+
```

Comment:
To me this looks slightly misleading - is being castable to a numpy int and actually being a numpy int the same thing, at least for what this is practically used for?

Reply:
with how the function is used yes, but it's a fair point on the function by itself. I'll rename to `_can_cast_to_numpy_int()`

Reply:
Ok sounds good! Happy for you to merge once done